cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/batch/preview/LimitingInputFormatProvider.java

```diff
@@ -34,7 +33,7 @@ public LimitingInputFormatProvider(InputFormatProvider delegate, int maxRecords)
 
   private static Map<String, String> getConfiguration(InputFormatProvider delegate, int maxRecords) {
     Map<String, String> config = new HashMap<>(delegate.getInputFormatConfiguration());
-    config.put(DelegatingInputFormat.DELEGATE_CLASS_NAME, delegate.getInputFormatClassName());
+    config.put(LimitingInputFormat.DELEGATE_CLASS_NAME, delegate.getInputFormatClassName());
     config.put(LimitingInputFormat.MAX_RECORDS, String.valueOf(maxRecords));
```

Comment:
I am wondering if we can use something like `ImmutableMap.Builder` here. It would immediately tell us if we have duplicate keys.

Reply:
We actually want to overwrite here (e.g. the MAX_RECORDS).